### PR TITLE
Add Verification-Scene to Dokan Target

### PR DESCRIPTION
**Pull request by ahmednaagy**

### CODE REVIEW

File: Dokan/Dokan/Scenes/Verification/VerificationViewController.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/Verification/VerificationViewController.swift]: You should use the current template

**Review by ahmdmhasn — APPROVED** [review]
The task should not be marked as done. This is an initial PR of multiple PRs to fully implement the feature.

**Review by RaniaZeid — COMMENTED** [review]
really great work

**Review by RaniaZeid — APPROVED** [review]

File: UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        }
+        
+        if text.count == digitsLabels.count {
+            didEnterLastDigit?(text)
+        }
+    }
+    
+}
+
+extension OTCTextField: UITextFieldDelegate {
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift:86]:
```suggestion
// MARK: - UITextFieldDelegate
//
extension OTCTextField: UITextFieldDelegate {
```

File: UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+        guard let charachterCount = textField.text?.count else { return false }
+        return charachterCount < digitsLabels.count || string == ""
+    }
+}
+
+
+
+
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift:95]: Please remove extra spaces. 
```suggestion
```

File: UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        
+        if text.count == digitsLabels.count {
+            didEnterLastDigit?(text)
+        }
+    }
+    
+}
+
+extension OTCTextField: UITextFieldDelegate {
+    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift:87]:
```suggestion

    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
```

File: UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            } else {
+                currentLabel.text = defaultCharacter
+            }
+        }
+        
+        if text.count == digitsLabels.count {
+            didEnterLastDigit?(text)
+        }
+    }
+
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift:83]:
```suggestion
```

File: UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+import UIKit
+
+class OTCTextField: UITextField {
+    
+    var didEnterLastDigit: ((String) -> Void)?
+    var defaultCharacter = "-"
+    private var isConfigured = false
+    private var digitsLabels = [UILabel]()
+    private lazy var tapRecognizer: UITapGestureRecognizer = {
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift:16]: No need to have this as instance property.

File: UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        addTarget(self, action: #selector(textDidChange), for: .editingChanged)
+        delegate = self
+    }
+    private func createLabelStackView(with count: Int) -> UIStackView {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .horizontal
+        stackView.alignment = .fill
+        stackView.distribution = .fillEqually
+        stackView.spacing = 8
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift:50]: Configuration which might change per class are preferably to be added in enum in the class 

For example
```swift 
extension OTCTextField { 
    
    enum Metrics { 
        static let labelsSpacing: CGFloat = 8.0
    }

    enum Style { 
        static let font: UIFont = .systemFont(ofSize: 40)
    }
}
```

Should be aded at the end of the file.

File: UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift [hunk 8]
```diff
@@ -0,0 +1,95 @@
+//
+//  OTCTextField.swift
+//  
+//
+//  Created by Ahmed Nagy on 08/07/2022.
+//
+
+import UIKit
+
+class OTCTextField: UITextField {
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift:10]: Please use decorator pattern to separate:
- stored properties
- Public handlers
- private handlers
- Configurations

File: UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    
+    var didEnterLastDigit: ((String) -> Void)?
+    var defaultCharacter = "-"
+    private var isConfigured = false
+    private var digitsLabels = [UILabel]()
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift:15]:
```suggestion
    private var digitsLabels: [UILabel] = []
```

File: UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .horizontal
+        stackView.alignment = .fill
+        stackView.distribution = .fillEqually
+        stackView.spacing = 8
+        (1...count).forEach { _ in
+            let label = UILabel()
+            label.translatesAutoresizingMaskIntoConstraints = false
+            label.textAlignment = .center
+            label.font = .systemFont(ofSize: 40)
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift:55]: Should be configurable using some sort of configurations.

File: UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+            label.isUserInteractionEnabled = true
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift:56]: Needed?

File: UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+        for i in 0..<digitsLabels.count {
+            let currentLabel = digitsLabels[i]
+            
+            if i < text.count {
+                let index = text.index(text.startIndex, offsetBy: i)
+                currentLabel.text = String(text[index])
+            } else {
+                currentLabel.text = defaultCharacter
+            }
+        }
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift:77]: Should be separate to separate method.

File: UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+        
+        if text.count == digitsLabels.count {
+            didEnterLastDigit?(text)
+        }
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/ReusableViews/OTCTextField.swift:81]: Should be separate to separate method.

**Review by ahmdmhasn — COMMENTED** [review]
Great effort. 
- Please check the comments and apply them in a new pull request.
- We need to integrate and finalize verification screen. 
@ahmednaagy @RaniaZeid

**Pull request merged**